### DISCUSSION TITLE
UI bug when trying to package it on Debian

### DISCUSSION THREAD

**Discussion by CicadaSeventeen** [BuddiesOfBudgie/budgie-screensaver#27] [General]
Everything works just fine when
```
    meson build
    meson compile -C build
    sudo meson install -C build

```

However, when I try to build and package it, which means like
```
    meson setup --prefix=/usr build
    meson compile -C build
    sudo meson install -C build --destdir xxx
```
There is a fatal bug that the login/unlock UI does not show up, so that I cannot input password.
What can be the problem?

**Comment by JoshStrobl** [comment]
You should not be changing destdir. That should remain the default, which ends up being `/` (effectively empty as root), which then adds the prefix (e.g. `/usr`) on top of it.

There is no UI bug when you are failing to build and install it properly.

**Comment by CicadaSeventeen** [comment]
> You should not be changing destdir. That should remain the default, which ends up being `/` (effectively empty as root), which then adds the prefix (e.g. `/usr`) on top of it.
> 
> There is no UI bug when you are failing to build and install it properly.

Excuse me. Maybe I did not explain clearly. The screensaver does not work correctly even when
```
    meson setup --prefix=/usr build
    meson compile -C build
    sudo meson install -C build
```
But it works when
```
    meson setup --prefix=/usr/libexec/test build
    meson compile -C build
    sudo meson install -C build
```
or
```
    meson setup build
    meson compile -C build
    sudo meson install -C build
```

The problem is that it does not work in `/usr` dir.

**Comment by fossfreedom** [comment]
Under debian you will need to recompile budgie desktop without using the meson option with-gnome-screensaver

You also obviously need to uninstall gnome-screensaver as well

**Comment by CicadaSeventeen** [comment]
> Under debian you will need to recompile budgie desktop without using the meson option with-gnome-screensaver
> 
> You also obviously need to uninstall gnome-screensaver as well

Do you mean under Debian or Ubuntu (Ubuntu has similar problem too), `budgie-core` needs to be recompile `budgie-desktop` if I want to switch to `budgie-screensaver` ?

**Comment by fossfreedom** [comment]
Both. You should uninstall budgie desktop, budgie core and all related libraries. Then compile budgie via apt-get source budgie-desktop
